pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -281,10 +323,52 @@ func manageRouteControllerManagerClientCA_v311_00_to_latest(client coreclientv1.
 	return resourceapply.SyncConfigMap(context.Background(), client, recorder, util.KubeAPIServerNamespace, apiserverClientCA, util.RouteControllerTargetNamespace, apiserverClientCA, []metav1.OwnerReference{})
 }
 
+// groupDisabledControllers gets caps from configv1.ClusterVersion and populates
+// groupDisabled with all controllers that are known in the ClusterVersion and
+// also disabled.
+func groupDisabledControllers(clusterVersion *configv1.ClusterVersion) map[controlplanev1.OpenShiftControllerName]bool {
+	knownCaps := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.KnownCapabilities...)
+	capsEnabled := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.EnabledCapabilities...)
+
+	groupDisabled := make(map[controlplanev1.OpenShiftControllerName]bool)
+	for _, cont := range knownControllers {
+		if controllerCapabilities[cont] != "" && knownCaps.Has(controllerCapabilities[cont]) && !capsEnabled.Has(controllerCapabilities[cont]) {
+			groupDisabled[cont] = true
+		}
+	}
+	return groupDisabled
+}
+
+// By default, all controllers are enabled. To disable specific controllers,
+// '-foo' disables the controller named 'foo'. For example, to disable the
+// OpenShiftServiceAccountController, one would use "-openshift.io/serviceaccount".
+func disableControllers(groupDisabled map[controlplanev1.OpenShiftControllerName]bool) []string {
```

Comment:
Given output of `groupDisabledControllers` is passed just into `disableControllers` both functions can be merged into a single one to avoid looping through `knownControllers` twice.